src/storage/heap_file.c

```diff
@@ -6817,28 +6817,12 @@ heap_scancache_start_modify (THREAD_ENTRY * thread_p, HEAP_SCANCACHE * scan_cach
 
       if (scan_cache->num_btids > 0)
 	{
-	  /* allocate local btree statistical information structure */
-	  malloc_size = sizeof (BTREE_UNIQUE_STATS) * scan_cache->num_btids;
-
-	  if (scan_cache->index_stat_info != NULL)
-	    {
-	      db_private_free (thread_p, scan_cache->index_stat_info);
-	    }
-
-	  scan_cache->index_stat_info = (BTREE_UNIQUE_STATS *) db_private_alloc (thread_p, malloc_size);
-	  if (scan_cache->index_stat_info == NULL)
-	    {
-	      ret = ER_OUT_OF_VIRTUAL_MEMORY;
-	      heap_classrepr_free_and_init (classrepr, &classrepr_cacheindex);
-	      goto exit_on_error;
-	    }
+	  delete scan_cache->m_index_stats;
```

Comment:
if (delete scan_cache->m_index_stats != NULL)

Reply:
Please clarify your suggestion.

If you mean to check pointer is not null before deleting, that is not necessary. deleting NULL pointer is no-op.

Reply:
ok